### ISSUE TITLE
Craft 4 - Migrations 'm220511_054742_delete_converted_field_block_types_and_groups' and  'm220516_124013_add_blocktype_description' are failing

### ISSUE THREAD

**Issue by maehnerm** [spicywebau/craft-neo#603]
#### Description
I'm trying to update from Craft 3 to Craft 4. But when I run `php ./craft migrate/all` it fails with the following console output:
```
Exception: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'description' in 'field list'
The SQL being executed was: SELECT `id`, `fieldId`, `fieldLayoutId`, `groupId`, `name`, `handle`, `description`, `maxBlocks`, `maxSiblingBlocks`, `maxChildBlocks`, `childBlocks`, `topLevel`, `sortOrder`, `uid`
FROM `craft_neoblocktypes`
WHERE `id`=81
ORDER BY `sortOrder`
LIMIT 1 (/app/vendor/yiisoft/yii2/db/Schema.php:676)
#0 /app/vendor/yiisoft/yii2/db/Command.php(1307): yii\db\Schema->convertException(Object(PDOException), 'SELECT `id`, `f...')
#1 /app/vendor/yiisoft/yii2/db/Command.php(1168): yii\db\Command->internalExecute('SELECT `id`, `f...')
#2 /app/vendor/yiisoft/yii2/db/Command.php(424): yii\db\Command->queryInternal('fetch', NULL)
#3 /app/vendor/yiisoft/yii2/db/Query.php(287): yii\db\Command->queryOne()
#4 /app/vendor/craftcms/cms/src/db/Query.php(180): yii\db\Query->one(NULL)
#5 /app/vendor/spicyweb/craft-neo/src/services/BlockTypes.php(61): craft\db\Query->one()
#6 /app/vendor/spicyweb/craft-neo/src/services/BlockTypes.php(481): benf\neo\services\BlockTypes->getById(81)
#7 /app/vendor/craftcms/cms/src/services/ProjectConfig.php(1116): benf\neo\services\BlockTypes->handleDeletedBlockType(Object(craft\events\ConfigEvent))
#8 [internal function]: craft\services\ProjectConfig->handleChangeEvent(Object(craft\events\ConfigEvent))
#9 /app/vendor/yiisoft/yii2/base/Component.php(633): call_user_func(Array, Object(craft\events\ConfigEvent))
#10 /app/vendor/craftcms/cms/src/models/ProjectConfigData.php(76): yii\base\Component->trigger('removeItem', Object(craft\events\ConfigEvent))
#11 /app/vendor/craftcms/cms/src/services/ProjectConfig.php(505): craft\models\ProjectConfigData->commitChanges(Array, NULL, 'neoBlockTypes.9...', true, NULL, true)
#12 /app/vendor/craftcms/cms/src/services/ProjectConfig.php(523): craft\services\ProjectConfig->set('neoBlockTypes.9...', NULL, NULL)
#13 /app/vendor/spicyweb/craft-neo/src/migrations/m220511_054742_delete_converted_field_block_types_and_groups.php(39): craft\services\ProjectConfig->remove('neoBlockTypes.9...')
#14 /app/vendor/craftcms/cms/src/db/Migration.php(49): benf\neo\migrations\m220511_054742_delete_converted_field_block_types_and_groups->safeUp()
#15 /app/vendor/yiisoft/yii2/console/controllers/BaseMigrateController.php(758): craft\db\Migration->up()
#16 /app/vendor/craftcms/cms/src/console/controllers/MigrateController.php(360): yii\console\controllers\BaseMigrateController->migrateUp('m220511_054742_...')
#17 [internal function]: craft\console\controllers\MigrateController->actionAll()
#18 /app/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#19 /app/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#20 /app/vendor/yiisoft/yii2/console/Controller.php(182): yii\base\Controller->runAction('all', Array)
#21 /app/vendor/yiisoft/yii2/base/Module.php(552): yii\console\Controller->runAction('all', Array)
#22 /app/vendor/yiisoft/yii2/console/Application.php(180): yii\base\Module->runAction('migrate/all', Array)
#23 /app/vendor/craftcms/cms/src/console/Application.php(90): yii\console\Application->runAction('migrate/all', Array)
#24 /app/vendor/yiisoft/yii2/console/Application.php(147): craft\console\Application->runAction('migrate/all', Array)
#25 /app/vendor/craftcms/cms/src/console/Application.php(121): yii\console\Application->handleRequest(Object(craft\console\Request))
#26 /app/vendor/yiisoft/yii2/base/Application.php(384): craft\console\Application->handleRequest(Object(craft\console\Request))
#27 /app/craft(24): yii\base\Application->run()
#28 {main}
*** failed to apply m220511_054742_delete_converted_field_block_types_and_groups (time: 0.018s)
```

If I create the column 'description' manually in the database and run `php ./craft migrate/all` again it fails on the next migration saying:
```
    > add column description string AFTER `handle` to table {{%neoblocktypes}} ...Exception: SQLSTATE[42S21]: Column already exists: 1060 Duplicate column name 'description'
The SQL being executed was: ALTER TABLE `craft_neoblocktypes` ADD `description` varchar(255) AFTER `handle` (/app/vendor/yiisoft/yii2/db/Schema.php:676)
#0 /app/vendor/yiisoft/yii2/db/Command.php(1307): yii\db\Schema->convertException(Object(PDOException), 'ALTER TABLE `cr...')
#1 /app/vendor/yiisoft/yii2/db/Command.php(1102): yii\db\Command->internalExecute('ALTER TABLE `cr...')
#2 /app/vendor/yiisoft/yii2/db/Migration.php(377): yii\db\Command->execute()
#3 /app/vendor/spicyweb/craft-neo/src/migrations/m220516_124013_add_blocktype_description.php(18): yii\db\Migration->addColumn('{{%neoblocktype...', 'description', Object(craft\db\mysql\ColumnSchemaBuilder))
#4 /app/vendor/craftcms/cms/src/db/Migration.php(49): benf\neo\migrations\m220516_124013_add_blocktype_description->safeUp()
#5 /app/vendor/yiisoft/yii2/console/controllers/BaseMigrateController.php(758): craft\db\Migration->up()
#6 /app/vendor/craftcms/cms/src/console/controllers/MigrateController.php(360): yii\console\controllers\BaseMigrateController->migrateUp('m220516_124013_...')
#7 [internal function]: craft\console\controllers\MigrateController->actionAll()
#8 /app/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#9 /app/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#10 /app/vendor/yiisoft/yii2/console/Controller.php(182): yii\base\Controller->runAction('all', Array)
#11 /app/vendor/yiisoft/yii2/base/Module.php(552): yii\console\Controller->runAction('all', Array)
#12 /app/vendor/yiisoft/yii2/console/Application.php(180): yii\base\Module->runAction('migrate/all', Array)
#13 /app/vendor/craftcms/cms/src/console/Application.php(90): yii\console\Application->runAction('migrate/all', Array)
#14 /app/vendor/yiisoft/yii2/console/Application.php(147): craft\console\Application->runAction('migrate/all', Array)
#15 /app/vendor/craftcms/cms/src/console/Application.php(121): yii\console\Application->handleRequest(Object(craft\console\Request))
#16 /app/vendor/yiisoft/yii2/base/Application.php(384): craft\console\Application->handleRequest(Object(craft\console\Request))
#17 /app/craft(24): yii\base\Application->run()
#18 {main}
*** failed to apply m220516_124013_add_blocktype_description (time: 0.009s)
```

If I delete the column 'description' manually in the database and run `php ./craft migrate/all` again it's completing the migration process successfully.

#### Steps to reproduce
1. Update composer.json:  `"craftcms/cms": "3.7.27.2",` to `"craftcms/cms": "^4.0"` and `"spicyweb/craft-neo": "2.13.3"` to `"spicyweb/craft-neo": "^3.1.0"`
2. Run `composer update`
3. Run `php ./craft migrate/all`

#### Other information
- Neo version: 3.1.0
- Craft version: 4.0.4
- (if multi-site) What is the affected Neo field's propagation method: It's a multi-site, but I don't know how to find that out.
- Is eager-loading used? I don't know how to find that out.

**Comment by ttempleton** [comment]
Thanks for reporting this. It turned out that this affected Craft installs where there was leftover Neo data from a Neo-to-Matrix conversion, since that's what the `m220511_054742_delete_converted_field_block_types_and_groups` migration is deleting. I've just fixed that for the next release.

**Comment by ttempleton** [comment]
Just released Neo 3.1.1 which includes the fix for this issue.
